lighthouse-core/audits/dobetterweb/no-vulnerable-libraries.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Audits a page to make sure there are no JS libraries with
+ * known vulnerabilities being used. Checks against a vulnerability db
+ * provided by Snyk.io and checked in locally as third-party/snyk-snapshot.json
+ */
+
+'use strict';
+
+const Audit = require('../audit');
+
+class NoVulnerableLibrariesAudit extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Security',
+      name: 'no-vulnerable-libraries',
+      description: 'Avoids front-end JavaScript libraries'
+        + ' with known security vulnerabilities',
+      helpText: 'Some third-party scripts may contain known security vulnerabilities ' +
+        ' that are easily identified and exploited by attackers.',
+      requiredArtifacts: ['JSVulnerableLibraries']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const libraries = artifacts.JSVulnerableLibraries;
+
+    const finalVulns = Object.assign(...libraries.filter(obj => {
+      return obj.vulns;
+    }).map(record => {
+      const libVulns = [];
+      for (const i in record.vulns) {
+        if (Object.hasOwnProperty.call(record.vulns, i)) {
+          libVulns.push(record.vulns[i]);
```

Comment:
I think markdown is supported here, so could you make the Snyk URLs clickable? Also, is it possible to see the vulnerability itself in the results?

For example:

Library | Vulnerability | Severity | More Info
-- | -- | -- | --
jQuery@1.12.4 | Cross Site Scripting (XSS) | medium | https://snyk.io/vuln/npm:jquery:20150627

Here's how it appears now:

![image](https://user-images.githubusercontent.com/1120896/29386038-244d31bc-82a8-11e7-9251-cea9b24593f1.png)

I also reordered the table so the lib is first but I nitpick.

Reply:
No generic markdown support but there's a `url` type in the report renderer that should be easy to add clickable support for :)

Reply:
Are you ok with me doing that on this PR? Just didn't know if I should break it out separately for any reason.

Reply:
It'd be super useful elsewhere too and is fairly standalone so might be easier to review in isolation 👍

Reply:
See https://github.com/GoogleChrome/lighthouse/pull/3154